[MSE/SourceBuffer] MediaSourcePrivate doesn't need to know about CDMInstance/CDMSession

jyavenard opened the pull request:
#### 85760d3ab1e8c259cb88e61891e9c37837752feb
<pre>
[MSE/SourceBuffer] MediaSourcePrivate doesn&apos;t need to know about CDMInstance/CDMSession
<a href="https://bugs.webkit.org/show_bug.cgi?id=301073">https://bugs.webkit.org/show_bug.cgi?id=301073</a>
<a href="https://rdar.apple.com/163005164">rdar://163005164</a>

Reviewed by Eric Carlson.

The MediaPlayerPrivateMediaSourceAVFObjC had to communicate to all the
SourceBufferPrivateAVFObjC to notify them if a new CDMInstance or LegacyCDMSession
was attached to the player.
Now that the AudioVideoRenderer handles the decryption of MediaSample
the MediaSourcePrivateAVFObjC and all attached SourceBufferPrivateAVFObjCs
do not need to be notified that a CDM has been attached. This knowledge
is the responsibility of the MediaPlayerPrivate and its AudioVideoRenderer.
We can remove all associated code.

No change in observable behaviour, covered by existing tests.

* Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm:
(WebCore::AudioVideoRendererAVFObjC::enqueueSample):
(WebCore::AudioVideoRendererAVFObjC::canEnqueueSample): Fix from 301391@main
didn&apos;t carry over in 301865@main. Causing seek with encrypted audio to never complete.
(WebCore::AudioVideoRendererAVFObjC::attachContentKeyToSampleIfNeeded):

Canonical link: <a href="https://commits.webkit.org/301921@main">https://commits.webkit.org/301921@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4abc053e305371e76e62b0f948a42823bad2d4b9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127495 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/47143 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38281 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134571 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/79051 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/8ce58c09-3d9a-4f1f-9ee7-1ce7cd9c6457) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/129367 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/47760 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/55667 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/97047 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/79051 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0383f769-53d0-4207-ad83-49b7774d6cae) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130443 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/47760 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/114176 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/77528 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/9b4f7539-9fd0-4699-8642-757550bed96d) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/47760 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/32280 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77945 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/47760 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/32704 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/137056 "Built successfully") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/54154 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/55667 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/105573 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/54665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110532 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/105224 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26823 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/29185 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/51733 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/54091 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/60178 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/53325 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/56782 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/55084 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm:
```diff
@@ -201,12 +206,14 @@ @interface AVSampleBufferAudioRenderer (WebCoreSampleBufferKeySession) <AVConten
     attachContentKeyToSampleIfNeeded(sample);
 #endif
 
+    ALWAYS_LOG(LOGIDENTIFIER, "enqueue sample: ", sample.get(), " for track: ", toString(trackId));
```
This is going to be really noise so maybe change it to a `DEBUG_LOG`?

jyavenard committed:
[MSE/SourceBuffer] MediaSourcePrivate doesn't need to know about CDMInstance/CDMSession
https://bugs.webkit.org/show_bug.cgi?id=301073
rdar://163005164

Reviewed by Eric Carlson.

The MediaPlayerPrivateMediaSourceAVFObjC had to communicate to all the
SourceBufferPrivateAVFObjC to notify them if a new CDMInstance or LegacyCDMSession
was attached to the player.
Now that the AudioVideoRenderer handles the decryption of MediaSample
the MediaSourcePrivateAVFObjC and all attached SourceBufferPrivateAVFObjCs
do not need to be notified that a CDM has been attached. This knowledge
is the responsibility of the MediaPlayerPrivate and its AudioVideoRenderer.
We can remove all associated code.

No change in observable behaviour, covered by existing tests.

* Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm:
(WebCore::AudioVideoRendererAVFObjC::enqueueSample):
(WebCore::AudioVideoRendererAVFObjC::canEnqueueSample): Fix from 301391@main
didn't carry over in 301865@main. Causing seek with encrypted audio to never complete.
(WebCore::AudioVideoRendererAVFObjC::attachContentKeyToSampleIfNeeded):

Canonical link: https://commits.webkit.org/301921@main
